File: Programmers/42578.py
# 해시 - 위장

from itertools import combinations


def solution(clothes):
    dress = {}
    for v, k in clothes:
        if k in dress:
            dress[k] += 1
        else:
            dress[k] = 1

    answer = 1

    for key in dress.keys():
        answer *= dress[key] + 1

    return answer - 1

File: Programmers/42578.js
// 해시 - 위장
function solution(clothes) {
    var answer = 1;
    let dress = {};
    clothes.forEach((cloth) => {
        if (dress[cloth[1]] === undefined) {
            dress[cloth[1]] = 1;
        } else {
            dress[cloth[1]]++;
        }
    });

    Object.keys(dress).forEach(element => {
        answer *= (dress[element] + 1);
    })
    return answer - 1;
}